src/Components/Facility/AssetCreate.tsx

```diff
@@ -0,0 +1,353 @@
+import React, { useReducer, useState, useEffect } from "react";
+import { createAsset, listFacilityAssetLocation } from "../../Redux/actions";
+import { useDispatch } from "react-redux";
+import * as Notification from "../../Utils/Notifications.js";
+import loadable from "@loadable/component";
+import CheckCircleOutlineIcon from "@material-ui/icons/CheckCircleOutline";
+import {
+  Button,
+  Card,
+  CardContent,
+  CircularProgress,
+  InputLabel,
+  IconButton,
+  RadioGroup,
+  FormControlLabel,
+  Box,
+  Radio,
+} from "@material-ui/core";
+import {
+  SelectField,
+  TextInputField,
+  MultilineInputField,
+} from "../Common/HelperInputFields";
+
+const PageTitle = loadable(() => import("../Common/PageTitle"));
+const initForm: any = {
+  name: "",
+  asset_type: "",
+  description: "",
+  is_working: "",
+  serial_number: "",
+  warranty_details: "",
+  location: "0",
+};
+
+const initError = Object.assign(
+  {},
+  ...Object.keys(initForm).map((k) => ({ [k]: "" }))
+);
+
+const initialState = {
+  form: { ...initForm },
+  errors: { ...initError },
+};
+
+interface AssetProps {
+  facilityId: string;
+}
+
+const asset_create_reducer = (state = initialState, action: any) => {
+  switch (action.type) {
+    case "set_form": {
+      return {
+        ...state,
+        form: action.form,
+      };
+    }
+    case "set_error": {
+      return {
+        ...state,
+        errors: action.errors,
+      };
+    }
+    default:
+      return state;
+  }
+};
```

Comment:
let's use multiple local states instead of using reducers.

Reply:
Reducers is being used it many part of the code especially when there is larger state to manage... I guess it works here fine.